### ISSUE TITLE
Update go-serial to build for darwin

### ISSUE THREAD

**Issue by dehanj** [tillitis/tillitis-key1-apps#103]
Due to stricter enforcement of rules in Go 1.21, go-serial version < 1.6.0 does not build on darwin. 
So for our applications to be supported on darwin for Go 1.21, go-serial needs to be updated to at > 1.6.0, see release notes [here](https://github.com/bugst/go-serial/releases/tag/v1.6.0).

```
# go.bug.st/serial/enumerator
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:142:11: cannot define new methods on non-local type C.CFStringRef
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:148:11: cannot define new methods on non-local type C.CFTypeRef
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:154:11: cannot define new methods on non-local type *C.io_registry_entry_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:165:11: cannot define new methods on non-local type *C.io_registry_entry_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:175:11: cannot define new methods on non-local type *C.io_registry_entry_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:194:11: cannot define new methods on non-local type *C.io_registry_entry_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:214:11: cannot define new methods on non-local type *C.io_iterator_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:218:11: cannot define new methods on non-local type *C.io_iterator_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:222:11: cannot define new methods on non-local type *C.io_iterator_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:229:11: cannot define new methods on non-local type *C.io_object_t
../../../go/pkg/mod/go.bug.st/serial@v1.5.0/enumerator/usb_darwin.go:233:11: cannot define new methods on non-local type *C.io_object_t
```

**Comment by dehanj** [comment]
close with #105

**Comment by stefanb** [comment]
This still needs to be released so that it can be built with Go 1.22:
* https://github.com/Homebrew/homebrew-core/pull/157782

**Comment by dehanj** [comment]
> This still needs to be released so that it can be built with Go 1.22:
> 
>     * [go 1.22rc1 Homebrew/homebrew-core#157782](https://github.com/Homebrew/homebrew-core/pull/157782)

Correct, we have a few more things we want to include before release.
